Compilation error with *clrIfExpanded directive and strictTemplates = true

## Describe the bug

When using the `*clrIfExpanded` directive as documented with strictTemplates check enabled in Angular 9/10, i get the following compilation error: 

`error TS2322: Type 'string' is not assignable to type 'boolean'`

## How to reproduce

Steps to reproduce the behavior:

1.  Add the *clrIfExpanded directive to any supported component (i had it on grid `clr-dg-row-detail`)
2.  Enable strict template checks (https://angular.io/guide/template-typecheck#strict-mode)
4.  Compile and see build error

## Expected behavior

The compiler throws no error with the documented approach

## Versions

**App**

- Angular: 10
- Node: 12.16
- Clarity: 3.1.4

--- a/packages/angular/golden/clr-angular.d.ts
+++ b/packages/angular/golden/clr-angular.d.ts
@@ -741,8 +741,8 @@ export declare class ClrDatagridRow<T = any> implements AfterContentInit, AfterV
     get clrDgDetailCloseLabel(): string;
     set clrDgDetailOpenLabel(label: string);
     get clrDgDetailOpenLabel(): string;
-    set clrDgSelectable(value: boolean);
-    get clrDgSelectable(): boolean;
+    set clrDgSelectable(value: boolean | string);
+    get clrDgSelectable(): boolean | string;
     commonStrings: ClrCommonStringsService;
     detailButton: HTMLButtonElement;
     detailService: DetailService;
@@ -752,8 +752,8 @@ export declare class ClrDatagridRow<T = any> implements AfterContentInit, AfterV
     expandAnimation: ClrExpandableAnimation;
     expandAnimationTrigger: boolean;
     expandableId: string;
-    get expanded(): boolean;
-    set expanded(value: boolean);
+    get expanded(): boolean | string;
+    set expanded(value: boolean | string);
     expandedChange: EventEmitter<boolean>;
     globalExpandable: ExpandableRowsCount;
     id: string;
@@ -761,8 +761,8 @@ export declare class ClrDatagridRow<T = any> implements AfterContentInit, AfterV
     radioId: string;
     replaced: boolean;
     rowActionService: RowActionService;
-    get selected(): boolean;
-    set selected(value: boolean);
+    get selected(): boolean | string;
+    set selected(value: boolean | string);
     selectedChanged: EventEmitter<boolean>;
     selection: Selection<T>;
     constructor(selection: Selection<T>, rowActionService: RowActionService, globalExpandable: ExpandableRowsCount, expand: DatagridIfExpandService, detailService: DetailService, displayMode: DisplayModeService, vcr: ViewContainerRef, renderer: Renderer2, el: ElementRef, commonStrings: ClrCommonStringsService);
@@ -1136,8 +1136,8 @@ export declare class ClrIfExpanded implements OnInit, OnDestroy {
 }
 
 export declare class ClrIfOpen implements OnDestroy {
-    set open(value: boolean);
-    get open(): boolean;
+    set open(value: boolean | string);
+    get open(): boolean | string;
     openChange: EventEmitter<boolean>;
     constructor(toggleService: ClrPopoverToggleService, template: TemplateRef<any>, container: ViewContainerRef);
     ngOnDestroy(): void;
@@ -1892,10 +1892,10 @@ export declare class ClrTreeViewModule {
 
 export declare class ClrVerticalNav implements OnDestroy {
     get ariaExpanded(): string;
-    get collapsed(): boolean;
-    set collapsed(value: boolean);
-    get collapsible(): boolean;
-    set collapsible(value: boolean);
+    get collapsed(): boolean | string;
+    set collapsed(value: boolean | string);
+    get collapsible(): boolean | string;
+    set collapsible(value: boolean | string);
     commonStrings: ClrCommonStringsService;
     get hasIcons(): boolean;
     get hasNavGroups(): boolean;
@@ -1911,7 +1911,7 @@ export declare class ClrVerticalNavGroup implements AfterContentInit, OnDestroy 
     get expanded(): boolean;
     set expanded(value: boolean);
     expandedChange: EventEmitter<boolean>;
-    set userExpandedInput(value: boolean);
+    set userExpandedInput(value: boolean | string);
     constructor(_itemExpand: IfExpandService, _navGroupRegistrationService: VerticalNavGroupRegistrationService, _navGroupService: VerticalNavGroupService, _navService: VerticalNavService, commonStrings: ClrCommonStringsService);
     collapseGroup(): void;
     expandAnimationDone($event: AnimationEvent): void;

--- a/packages/angular/projects/clr-angular/src/data/datagrid/datagrid-row.ts
+++ b/packages/angular/projects/clr-angular/src/data/datagrid/datagrid-row.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -124,17 +124,17 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit 
   }
 
   @Input('clrDgSelected')
-  public set selected(value: boolean) {
+  public set selected(value: boolean | string) {
     if (this.selection.selectionType === SelectionType.None) {
-      this._selected = value;
+      this._selected = value as boolean;
     } else {
-      this.selection.setSelected(this.item, value);
+      this.selection.setSelected(this.item, value as boolean);
     }
   }
 
   // By default every item is selectable; it becomes not selectable only if it's explicitly set to false
   @Input('clrDgSelectable')
-  public set clrDgSelectable(value: boolean) {
+  public set clrDgSelectable(value: boolean | string) {
     this.selection.lockItem(this.item, value === false);
   }
 
@@ -156,8 +156,8 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit 
   }
 
   @Input('clrDgExpanded')
-  public set expanded(value: boolean) {
-    this.expand.expanded = value;
+  public set expanded(value: boolean | string) {
+    this.expand.expanded = value as boolean;
   }
 
   @Output('clrDgExpandedChange') expandedChange = new EventEmitter<boolean>(false);

--- a/packages/angular/projects/clr-angular/src/layout/vertical-nav/vertical-nav-group.ts
+++ b/packages/angular/projects/clr-angular/src/layout/vertical-nav/vertical-nav-group.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -99,7 +99,7 @@ export class ClrVerticalNavGroup implements AfterContentInit, OnDestroy {
   }
 
   @Input('clrVerticalNavGroupExpanded')
-  set userExpandedInput(value: boolean) {
+  set userExpandedInput(value: boolean | string) {
     value = !!value;
     if (this.expanded !== value) {
       // We have to call toggleExpand because some cases require animations to occur first

--- a/packages/angular/projects/clr-angular/src/layout/vertical-nav/vertical-nav.ts
+++ b/packages/angular/projects/clr-angular/src/layout/vertical-nav/vertical-nav.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -24,22 +24,22 @@ import { ClrCommonStringsService } from '../../utils/i18n/common-strings.service
   },
 })
 export class ClrVerticalNav implements OnDestroy {
-  get collapsible(): boolean {
+  get collapsible(): boolean | string {
     return this._navService.collapsible;
   }
 
   @Input('clrVerticalNavCollapsible')
-  set collapsible(value: boolean) {
-    this._navService.collapsible = value;
+  set collapsible(value: boolean | string) {
+    this._navService.collapsible = value as boolean;
   }
 
-  get collapsed(): boolean {
+  get collapsed(): boolean | string {
     return this._navService.collapsed;
   }
 
   @Input('clrVerticalNavCollapsed')
-  set collapsed(value: boolean) {
-    this._navService.collapsed = value;
+  set collapsed(value: boolean | string) {
+    this._navService.collapsed = value as boolean;
   }
 
   @Output('clrVerticalNavCollapsedChange')

--- a/packages/angular/projects/clr-angular/src/utils/conditional/if-open.directive.ts
+++ b/packages/angular/projects/clr-angular/src/utils/conditional/if-open.directive.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -33,8 +33,8 @@ export class ClrIfOpen implements OnDestroy {
    * @param value
    */
   @Input('clrIfOpen')
-  public set open(value: boolean) {
-    this.toggleService.open = value;
+  public set open(value: boolean | string) {
+    this.toggleService.open = value as boolean;
   }
 
   /********

